quiz/src/components/QuizForm.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useEffect } from "react";
+
+function Quizform({ questions }) {
+  const [answer, setAnswer] = useState("");
+  const [answers, setAnswers] = useState([
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+  ]);
+  const [result, setResult] = useState(0);
+  const [display, setDisplay] = useState(true);
+
+  useEffect(() => {
+    for (let i = 0; i <= 10; i++) {
+      const inputs = document.querySelectorAll(`.questions${i}`);
+      inputs.forEach((e) => {
+        if (e.checked) {
+          e.parentNode.style.background = 'rgb(114, 52, 95)';
+          e.parentNode.style.color = "white";
+
+        } else {
+          e.parentNode.style.background = "white";
+          e.parentNode.style.color = "black";
+        }
+      });
+    }
+  }, [answers]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setDisplay(false);
+    setResult(answers.filter((e) => e === "correct").length);
+  };
+
+  const handleChange = (e, index) => {
+    setAnswer(e.target.value);
+    setAnswers([
+      ...answers.slice(0, index),
+      e.target.value,
+      ...answers.slice(index + 1),
+    ]);
+  };
+  return (
+    <>
+      {display && (
+        <form onSubmit={handleSubmit}>
+          {questions.map((e, index) => (
+            <div key={index}>
```

Comment:
fix the case when questions array is empty , you can show a message with " no questions for this category "